src/traversals/diffusion.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    diffusion(g, p, n)
+
+Run diffusion simulation on `g` for `n` steps with spread
+probabilities based on `p`. Return a vector with the set of vertices
+reached at each step of the simulation.
+
+### Optional Arguments
+- `initial_infections=sample(vertices(g), 1)`: A list of vertices that
+are infected at the start of the simulation.
+- `watch=Set()`: While simulation is always run on the full graph,
+specifying `watch` limits reporting to a specific set of vertices reached
+during the simulation. If left as an empty set, all vertices will be watched.
+- `normalize=false`: if `false`, set the probability of spread from a vertex ``i`` to
+each of the out_neighbors of ``i`` to ``p``. If `true`, set the probability of spread
+from a vertex ``i`` to each of the `out_neighbors` of ``i`` to
+``\\frac{p}{outdegreee(g, i)}``.
+"""
+
+function diffusion(g::AbstractGraph,
+                   p::AbstractFloat,
```

Comment:
This should probably just be `Real`. You can use rational numbers in `randsubseq`.